swervelib/ctre/CanCoderFactoryBuilder.java

```diff
@@ -1,70 +1,91 @@
-import com.ctre.phoenix.sensors.WPI_CANCoder;
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.MagnetSensorConfigs;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.signals.AbsoluteSensorRangeValue;
+import com.ctre.phoenix6.signals.SensorDirectionValue;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import frc.team670.mustanglib.swervelib.AbsoluteEncoder;
 import frc.team670.mustanglib.swervelib.AbsoluteEncoderFactory;
 
 public class CanCoderFactoryBuilder {
 
-    private Direction direction = Direction.COUNTER_CLOCKWISE;
     private int periodMilliseconds = 10;
 
     public CanCoderFactoryBuilder withReadingUpdatePeriod(int periodMilliseconds) {
         this.periodMilliseconds = periodMilliseconds;
         return this;
     }
 
-    public CanCoderFactoryBuilder withDirection(Direction direction) {
-        this.direction = direction;
-        return this;
-    }
-
     public AbsoluteEncoderFactory<CanCoderAbsoluteConfiguration> build() {
         return configuration -> {
-            CANCoderConfiguration config = new CANCoderConfiguration();
-            config.absoluteSensorRange = AbsoluteSensorRange.Unsigned_0_to_360;
-            config.magnetOffsetDegrees = Math.toDegrees(configuration.getOffset());
-            config.sensorDirection = direction == Direction.CLOCKWISE;
-            config.initializationStrategy = configuration.getInitStrategy();
+            CANcoderConfiguration config = new CANcoderConfiguration();
+            CANcoder encoder = new CANcoder(configuration.getId(), configuration.getCanbus());
+            
+            
+            StatusCode val = encoder.getConfigurator().refresh(config);
+
+            if(val.isOK()){
+                  val = encoder.getConfigurator().refresh(config);
+            }
+
+            
+            config.MagnetSensor.AbsoluteSensorRange = AbsoluteSensorRangeValue.Unsigned_0To1;
+            // config.MagnetSensor.MagnetOffset = configuration.getOffset();// Should be roations
+            config.MagnetSensor.SensorDirection = SensorDirectionValue.CounterClockwise_Positive;
 
-            WPI_CANCoder encoder = new WPI_CANCoder(configuration.getId(), configuration.getCanbus());
-            CtreUtils.checkCtreError(encoder.configAllSettings(config, 250), "Failed to configure CANCoder");
+            encoder.getConfigurator().apply(config);
+            CtreUtils.checkCtreError(encoder.getConfigurator().apply(config), "Failed to configure CANCoder");
 
-            CtreUtils.checkCtreError(encoder.setStatusFramePeriod(CANCoderStatusFrame.SensorData, periodMilliseconds, 250), "Failed to configure CANCoder update rate");
+            CtreUtils.checkCtreError(encoder.optimizeBusUtilization(periodMilliseconds), "Failed to configure CANCoder update rate");
```

Comment:
I don't think this is quite right. I believe this will disable all cancoder updates. Which might be ok to do later but probably not right now. See https://pro.docs.ctr-electronics.com/en/stable/docs/api-reference/api-usage/status-signals.html for documentation on this stuff. I think we want something like

get the absolute encoder signal from https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/core/CoreCANcoder.html#getAbsolutePosition()
and then call
https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/BaseStatusSignal.html#setUpdateFrequency(double)
on that signal object, which should set the frequency for the signal. And after that I think it's ok to call optimizeBusUtilization to drop all other signals since all we care about is the absolute position signal. I think. This definitely all needs to be tested.

It's worth your time to read through that Status Signals documentation.